### PR TITLE
Add sonneityping

### PR DESCRIPTION
**Pull request by emarinier**
FOR CONTRIBUTOR:
* [x] - I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [x] - License permits unrestricted use (educational + commercial)
* [x] - This PR adds a new tool or tool collection
* [ ] - This PR updates an existing tool or tool collection
* [ ] - This PR does something else (explain below)

This tool adds the parse_mykrobe_predict functionality of [sonneityping](https://github.com/katholt/sonneityping).

### CODE REVIEW

File: tools/sonneityping/parse_mykrobe_predict.xml [hunk 1]
```diff
@@ -0,0 +1,69 @@
+<tool id="parse_mykrobe_predict" name="parse mykrobe predict" version="@TOOL_VERSION@">
+    <description>Parses Mykrobe predict results for Shigella sonnei.</description>
```
Comment (bgruening) [inline tools/sonneityping/parse_mykrobe_predict.xml:2]: The name and description is a little bit redundant. Can you harmonize it?

File: tools/sonneityping/parse_mykrobe_predict.xml [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    <macros>
+        <import>macros.xml</import>
+    </macros>
+    <expand macro="requirements"/>
+    <command detect_errors="exit_code"><![CDATA[
+        parse_mykrobe_predict --jsons
+
+        #for $i in $json
+            #if $i
+                "$i"
```
Comment (bgruening) [inline tools/sonneityping/parse_mykrobe_predict.xml:12]: Please use single quotes. This is more secure.

File: tools/sonneityping/parse_mykrobe_predict.xml [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+        #if $optional.alleles:
+            --alleles '$optional.alleles'
+        #end if
+
+        --prefix results
+
+    ]]></command>
+    <inputs>
+        <param name="json" type="data" format="json" multiple="true" optional="false" label="The Mykrobe results to parse." help="The Mykrobe results, in JSON format, to parse. This input is mandatory." />
```
Comment (bgruening) [inline tools/sonneityping/parse_mykrobe_predict.xml:24]: You don't need to say that the input is mandatory in the help.. the leading . is also not necessary.

File: tools/sonneityping/parse_mykrobe_predict.xml [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+
+        <section name="optional" title="Optional Parameters" expanded="false">
+            <param name="alleles" type="data" format="text,tabular" optional="true" label="Alleles" help="Links each lineage name to its human-readable counterpart. This parameter is optional." />
```
Comment (bgruening) [inline tools/sonneityping/parse_mykrobe_predict.xml:27]: txt and tabular.

File: tools/sonneityping/parse_mykrobe_predict.xml [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+        </section>
+    </inputs>
+    <outputs>
+        <data format="tabular" label="parsed output" name="parsed_output" from_work_dir="results_predictResults.tsv" />
+    </outputs>
+    <tests>
+        <test>
+            <param name="json" ftype="json" value="SRR6114360.json"/>
+            <param name="optional.alleles" ftype="text" value="alleles.txt" />
```
Comment (bgruening) [inline tools/sonneityping/parse_mykrobe_predict.xml:36]: txt

**Review by bgruening — COMMENTED** [review]
Thanks! Small comments inline :-)

File: tools/sonneityping/parse_mykrobe_predict.xml (hunk 1 above)
Comment (emarinier) [reply tools/sonneityping/parse_mykrobe_predict.xml]: I've update the description to:

"Parses JSON files output by Mykrobe for Shigella sonnei and tabulates results into a single tab-delimited file."

Let me know if this isn't what you had in mind.

File: tools/sonneityping/parse_mykrobe_predict.xml (hunk 2 above)
Comment (emarinier) [reply tools/sonneityping/parse_mykrobe_predict.xml]: Updated

File: tools/sonneityping/parse_mykrobe_predict.xml (hunk 3 above)
Comment (emarinier) [reply tools/sonneityping/parse_mykrobe_predict.xml]: Changed to:

help="Mykrobe predict results in JSON format."

File: tools/sonneityping/parse_mykrobe_predict.xml (hunk 4 above)
Comment (emarinier) [reply tools/sonneityping/parse_mykrobe_predict.xml]: Updated

File: tools/sonneityping/parse_mykrobe_predict.xml (hunk 5 above)
Comment (emarinier) [reply tools/sonneityping/parse_mykrobe_predict.xml]: Updated

**Comment by bgruening** [comment]
@emarinier do you have time to look at the last recommendations, or should I change that real quick?

**Comment by emarinier** [comment]
Sorry, I completely missed these requests. I'll make the changes soon!

File: tools/sonneityping/parse_mykrobe_predict.xml [hunk 6]
```diff
@@ -0,0 +1,69 @@
+<tool id="parse_mykrobe_predict" name="parse mykrobe predict" version="@TOOL_VERSION@" profile="20.05">
```
[the remark this answers is not in the document]
Comment (emarinier) [reply tools/sonneityping/parse_mykrobe_predict.xml]: Changed to `sonneityping`.

File: tools/sonneityping/parse_mykrobe_predict.xml [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+    <description>for Shigella sonnei and tabulate results into a single tab-delimited file</description>
+    <macros>
+        <import>macros.xml</import>
+    </macros>
+    <expand macro="requirements"/>
+    <command detect_errors="exit_code"><![CDATA[
+        parse_mykrobe_predict --jsons
+
+        #for $i in $json
+            #if $i
```
[the remark this answers is not in the document]
Comment (emarinier) [reply tools/sonneityping/parse_mykrobe_predict.xml]: Removed.

File: tools/sonneityping/parse_mykrobe_predict.xml [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+                    <has_text text="S. sonnei" />
+                </assert_contents>
+            </output>
+        </test>
+    </tests>
+    <help><![CDATA[
+Parse Mykrobe Predict
+=====================
+
+Parses Mykrobe predict results for Shigella sonnei. Please refer to the GitHub repository (https://github.com/katholt/sonneityping) for more information.]]></help>
```
[the remark this answers is not in the document]
Comment (emarinier) [reply tools/sonneityping/parse_mykrobe_predict.xml]: Updated as above.

**Review by bgruening — APPROVED** [review]

**Comment by bgruening** [comment]
@emarinier that is pretty cool. Thanks a lot!

**Pull request merged**